Add ability to call interactive on bound functions

philippjfr opened the pull request:
Implements functionality requested in https://github.com/holoviz/hvplot/issues/673

```python
from bokeh import sampledata

ticker = pn.widgets.Select(options=['AAPL', 'IBM', 'GOOG', 'MSFT'], name='Ticker')

def stock_df(ticker):
    df = pd.DataFrame(getattr(sampledata.stocks, ticker))
    df['date'] = pd.to_datetime(df.date)
    return df

stock_dfi = hvplot.bind(stock_df, ticker).interactive(width=600)

stock_dfi.head(10)
```

![interactive_df_select](https://user-images.githubusercontent.com/1550771/159768870-eed8f63a-5c40-49c7-8302-cd09f736d589.gif)

- [x] Add tests
- [x] Automatically detect whether to use DataFrameInteractive or XArrayInteractive object

jbednar reviewed (approved):
Looks great! Thanks for doing this.

MarcSkovMadsen commented:
Love this

philippjfr commented:
@maximlt Please review.

maximlt commented on examples/user_guide/Interactive.ipynb:
```diff
@@ -92,6 +123,68 @@
+    "\n",
+    "stock_dfi = hvplot.bind(stock_df, ticker).interactive(width=600)\n",
+    "\n",
+    "stock_dfi.head(10)"
```
I get an error when I run this example and select another company:
```python
Traceback (most recent call last):
  File "/Users/mliquet/miniconda3/envs/hvplot-dev37/lib/python3.7/site-packages/pyviz_comms/__init__.py", line 325, in _handle_msg
    self._on_msg(msg)
  File "/Users/mliquet/work/dev/panel/panel/viewable.py", line 274, in _on_msg
    doc.unhold()
  File "/Users/mliquet/miniconda3/envs/hvplot-dev37/lib/python3.7/site-packages/bokeh/document/document.py", line 799, in unhold
    self.callbacks.unhold()
  File "/Users/mliquet/miniconda3/envs/hvplot-dev37/lib/python3.7/site-packages/bokeh/document/callbacks.py", line 396, in unhold
    self.trigger_on_change(event)
  File "/Users/mliquet/miniconda3/envs/hvplot-dev37/lib/python3.7/site-packages/bokeh/document/callbacks.py", line 373, in trigger_on_change
    invoke_with_curdoc(doc, event.callback_invoker)
  File "/Users/mliquet/miniconda3/envs/hvplot-dev37/lib/python3.7/site-packages/bokeh/document/callbacks.py", line 408, in invoke_with_curdoc
    return f()
  File "/Users/mliquet/miniconda3/envs/hvplot-dev37/lib/python3.7/site-packages/bokeh/util/callback_manager.py", line 191, in invoke
    callback(attr, old, new)
  File "/Users/mliquet/work/dev/panel/panel/reactive.py", line 318, in _comm_change
    self._schedule_change(doc, comm)
  File "/Users/mliquet/work/dev/panel/panel/reactive.py", line 305, in _schedule_change
    self._change_event(doc)
  File "/Users/mliquet/work/dev/panel/panel/reactive.py", line 299, in _change_event
    self._process_events(events)
  File "/Users/mliquet/work/dev/panel/panel/reactive.py", line 265, in _process_events
    self.param.update(**self_events)
  File "/Users/mliquet/miniconda3/envs/hvplot-dev37/lib/python3.7/site-packages/param/parameterized.py", line 1873, in update
    setattr(self_or_cls, k, v)
  File "/Users/mliquet/miniconda3/envs/hvplot-dev37/lib/python3.7/site-packages/param/parameterized.py", line 354, in _f
    instance_param.__set__(obj, val)
  File "/Users/mliquet/miniconda3/envs/hvplot-dev37/lib/python3.7/site-packages/param/parameterized.py", line 356, in _f
    return f(self, obj, val)
  File "/Users/mliquet/miniconda3/envs/hvplot-dev37/lib/python3.7/site-packages/param/parameterized.py", line 1231, in __set__
    obj.param._call_watcher(watcher, event)
  File "/Users/mliquet/miniconda3/envs/hvplot-dev37/lib/python3.7/site-packages/param/parameterized.py", line 2016, in _call_watcher
    if watcher not in self_._watchers:
  File "/Users/mliquet/work/dev/hvplot/hvplot/interactive.py", line 293, in __eq__
    return self._apply_operator(operator.eq, other)
  File "/Users/mliquet/work/dev/hvplot/hvplot/interactive.py", line 261, in _apply_operator
    return self._clone(transform)
  File "/Users/mliquet/work/dev/hvplot/hvplot/interactive.py", line 166, in _clone
    loc=loc, center=center, dmap=dmap, **kwargs)
  File "/Users/mliquet/work/dev/hvplot/hvplot/interactive.py", line 114, in __init__
    self._current = self._transform.apply(ds, keep_index=True, compute=False)
  File "/Users/mliquet/work/dev/holoviews/holoviews/util/transform.py", line 739, in apply
    kwargs, accessor, drange)
  File "/Users/mliquet/work/dev/holoviews/holoviews/util/transform.py", line 638, in _apply_fn
    data = fn(*args, **kwargs)
  File "/Users/mliquet/miniconda3/envs/hvplot-dev37/lib/python3.7/site-packages/pandas/core/ops/common.py", line 69, in new_method
    return method(self, other)
  File "/Users/mliquet/miniconda3/envs/hvplot-dev37/lib/python3.7/site-packages/pandas/core/arraylike.py", line 32, in __eq__
    return self._cmp_method(other, operator.eq)
  File "/Users/mliquet/miniconda3/envs/hvplot-dev37/lib/python3.7/site-packages/pandas/core/frame.py", line 6851, in _cmp_method
    self, other = ops.align_method_FRAME(self, other, axis, flex=False, level=None)
  File "/Users/mliquet/miniconda3/envs/hvplot-dev37/lib/python3.7/site-packages/pandas/core/ops/__init__.py", line 289, in align_method_FRAME
    "Can only compare identically-labeled DataFrame objects"
ValueError: Can only compare identically-labeled DataFrame objects
```

philippjfr replied:
Am not able to reproduce this one.

maximlt replied:
This is what I have in the env causing the error:
```
# Name                    Version                   Build  Channel
geopandas                 0.9.0                      py_1  
geopandas-base            0.9.0                      py_1  
intake-xarray             0.5.0              pyhd3eb1b0_0  
numpy                     1.19.2           py37h63973fd_0  
numpy-base                1.19.2           py37h68fea81_0  
pandas                    1.3.4            py37h743cdd8_0  
spatialpandas             0.4.3                      py_0    pyviz/label/dev
xarray                    0.20.1             pyhd3eb1b0_1  
```

maximlt replied:
I've created a completely new environment on Python 3.7 and got the exact same error :/

<details>

  <summary>Output of conda list</summary>

```
# Name                    Version                   Build  Channel
abseil-cpp                20210324.2           he49afe7_0    conda-forge
affine                    2.3.1              pyhd8ed1ab_0    conda-forge
aiobotocore               2.1.2              pyhd8ed1ab_0    conda-forge
aiohttp                   3.8.1            py37h271585c_0    conda-forge
aioitertools              0.10.0             pyhd8ed1ab_0    conda-forge
aiosignal                 1.2.0              pyhd8ed1ab_0    conda-forge
anyio                     3.5.0            py37hf985489_0    conda-forge
appdirs                   1.4.4              pyh9f0ad1d_0    conda-forge
appnope                   0.1.2            py37hf985489_2    conda-forge
argon2-cffi               21.3.0             pyhd8ed1ab_0    conda-forge
argon2-cffi-bindings      21.2.0           py37h271585c_1    conda-forge
arrow-cpp                 7.0.0           py37h04e314c_2_cpu    conda-forge
asciitree                 0.3.3                      py_2    conda-forge
async-timeout             4.0.2              pyhd8ed1ab_0    conda-forge
async_generator           1.10                       py_0    conda-forge
asynctest                 0.13.0                     py_0    conda-forge
attrs                     21.4.0             pyhd8ed1ab_0    conda-forge
aws-c-cal                 0.5.11               hd2e2f4b_0    conda-forge
aws-c-common              0.6.2                h0d85af4_0    conda-forge
aws-c-event-stream        0.2.7               hb9330a7_13    conda-forge
aws-c-io                  0.10.5               h35aa462_0    conda-forge
aws-checksums             0.1.11               h0010a65_7    conda-forge
aws-sdk-cpp               1.8.186              h766a74d_3    conda-forge
babel                     2.9.1              pyh44b312d_0    conda-forge
backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
backports                 1.1                pyhd3eb1b0_0
backports.functools_lru_cache 1.6.4              pyhd8ed1ab_0    conda-forge
beautifulsoup4            4.10.0             pyha770c72_0    conda-forge
bleach                    4.1.0              pyhd8ed1ab_0    conda-forge
blosc                     1.21.0               he49afe7_0    conda-forge
bokeh                     2.4.2            py37hf985489_0    conda-forge
boost-cpp                 1.74.0               hdbf7018_7    conda-forge
botocore                  1.23.24            pyhd8ed1ab_0    conda-forge
branca                    0.4.2              pyhd8ed1ab_0    conda-forge
brotli                    1.0.9                h0d85af4_6    conda-forge
brotli-bin                1.0.9                h0d85af4_6    conda-forge
brotlipy                  0.7.0           py37h271585c_1003    conda-forge
brunsli                   0.1                  h046ec9c_0    conda-forge
bzip2                     1.0.8                h0d85af4_4    conda-forge
c-ares                    1.18.1               h0d85af4_0    conda-forge
c-blosc2                  2.0.4                ha1a4663_1    conda-forge
ca-certificates           2022.3.18            hecd8cb5_0
cairo                     1.16.0            h9e0e54b_1010    conda-forge
cartopy                   0.20.2           py37h73200b8_0    conda-forge
certifi                   2021.10.8        py37hf985489_1    conda-forge
cffi                      1.15.0           py37hc55c11b_1
cfitsio                   4.0.0                hb20e66c_0    conda-forge
cftime                    1.6.0            py37h4105427_0    conda-forge
chardet                   4.0.0            py37hf985489_2    conda-forge
charls                    2.3.4                he49afe7_0    conda-forge
charset-normalizer        2.0.12             pyhd8ed1ab_0    conda-forge
click                     8.1.0            py37hf985489_0    conda-forge
click-plugins             1.1.1                      py_0    conda-forge
cligj                     0.7.2              pyhd8ed1ab_1    conda-forge
cloudpickle               2.0.0              pyhd8ed1ab_0    conda-forge
codecov                   2.1.11             pyhd3deb0d_0    conda-forge
colorama                  0.4.4              pyh9f0ad1d_0    conda-forge
colorcet                  3.0.0                      py_0    pyviz/label/dev
conda                     4.12.0           py37hf985489_0    conda-forge
conda-build               3.21.8           py37hf985489_0    conda-forge
conda-package-handling    1.8.0            py37h69ee0a8_0    conda-forge
coverage                  6.3.2            py37h69ee0a8_1    conda-forge
cramjam                   2.5.0            py37h5210ebb_0    conda-forge
cryptography              36.0.0           py37hf6deb26_0
curl                      7.82.0               h9f20792_0    conda-forge
cycler                    0.11.0             pyhd8ed1ab_0    conda-forge
cytoolz                   0.11.2           py37h271585c_1    conda-forge
dask                      2022.2.0           pyhd8ed1ab_0    conda-forge
dask-core                 2022.2.0           pyhd8ed1ab_0    conda-forge
datashader                0.13.0                     py_0    pyviz/label/dev
datashape                 0.5.4                      py_1    conda-forge
dbus                      1.13.18              h18a8e69_0
debugpy                   1.5.1            py37hd8d24ac_0    conda-forge
decorator                 5.1.1              pyhd8ed1ab_0    conda-forge
defusedxml                0.7.1              pyhd8ed1ab_0    conda-forge
distributed               2022.2.0         py37hf985489_0    conda-forge
doit                      0.34.2           py37hf985489_0    conda-forge
entrypoints               0.4                pyhd8ed1ab_0    conda-forge
expat                     2.4.8                h96cf925_0    conda-forge
fasteners                 0.17.3             pyhd8ed1ab_0    conda-forge
fastparquet               0.8.0            py37h032687b_1    conda-forge
filelock                  3.6.0              pyhd8ed1ab_0    conda-forge
fiona                     1.8.20           py37hb742350_4    conda-forge
flake8                    4.0.1              pyhd8ed1ab_2    conda-forge
flit-core                 3.7.1              pyhd8ed1ab_0    conda-forge
folium                    0.12.1.post1       pyhd8ed1ab_1    conda-forge
font-ttf-dejavu-sans-mono 2.37                 hab24e00_0    conda-forge
font-ttf-inconsolata      3.000                h77eed37_0    conda-forge
font-ttf-source-code-pro  2.038                h77eed37_0    conda-forge
font-ttf-ubuntu           0.83                 hab24e00_0    conda-forge
fontconfig                2.13.96              h676cef8_2    conda-forge
fonts-conda-ecosystem     1                             0    conda-forge
fonts-conda-forge         1                             0    conda-forge
fonttools                 4.31.2           py37h69ee0a8_0    conda-forge
freetype                  2.11.0               hd8bbffd_0
freexl                    1.0.6                h0d85af4_0    conda-forge
frozenlist                1.3.0            py37h271585c_0    conda-forge
fsspec                    2022.2.0           pyhd8ed1ab_0    conda-forge
gdal                      3.4.0           py37h005c65d_12    conda-forge
geopandas                 0.10.2             pyhd8ed1ab_1    conda-forge
geopandas-base            0.10.2             pyha770c72_1    conda-forge
geos                      3.10.1               he49afe7_1    conda-forge
geotiff                   1.7.0                h9e3163f_5    conda-forge
geoviews                  1.9.5                      py_0    pyviz/label/dev
geoviews-core             1.9.5                      py_0    pyviz/label/dev
gettext                   0.21.0               h7535e17_0
gflags                    2.2.2             hb1e8313_1004    conda-forge
giflib                    5.2.1                hbcb3906_2    conda-forge
glib                      2.70.2               hcf210ce_4    conda-forge
glib-tools                2.70.2               hcf210ce_4    conda-forge
glob2                     0.7                        py_0    conda-forge
glog                      0.5.0                h25b26a9_0    conda-forge
grpc-cpp                  1.43.2               hb6405b5_1    conda-forge
h11                       0.13.0             pyhd8ed1ab_0    conda-forge
hdf4                      4.2.15               hefd3b78_3    conda-forge
hdf5                      1.12.1          nompi_ha60fbc9_104    conda-forge
heapdict                  1.0.1                      py_0    conda-forge
holoviews                 1.14.9a1                   py_0    pyviz/label/dev
hvplot                    0.8.0a12.post67+g3709775           dev_0    <develop>
icu                       69.1                 he49afe7_0    conda-forge
idna                      3.3                pyhd8ed1ab_0    conda-forge
imagecodecs               2021.11.20       py37h04bde26_2    conda-forge
imageio                   2.16.1             pyhcf75d05_0    conda-forge
importlib-metadata        4.11.3           py37hf985489_0    conda-forge
importlib_metadata        4.11.3               hd8ed1ab_0    conda-forge
importlib_resources       5.6.0              pyhd8ed1ab_0    conda-forge
iniconfig                 1.1.1              pyh9f0ad1d_0    conda-forge
intake                    0.6.5              pyhd8ed1ab_0    conda-forge
intake-parquet            0.2.3                      py_0    conda-forge
intake-xarray             0.6.0              pyhd8ed1ab_0    conda-forge
ipykernel                 6.10.0           py37h0a7177a_0    conda-forge
ipython                   7.32.0           py37hf985489_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
ipywidgets                7.7.0              pyhd8ed1ab_0    conda-forge
jbig                      2.1               h0d85af4_2003    conda-forge
jedi                      0.18.1           py37hf985489_0    conda-forge
jinja2                    3.1.1              pyhd8ed1ab_0    conda-forge
jmespath                  0.10.0             pyh9f0ad1d_0    conda-forge
joblib                    1.1.0              pyhd8ed1ab_0    conda-forge
jpeg                      9e                   h0d85af4_0    conda-forge
json-c                    0.15                 hcb556a6_0    conda-forge
json5                     0.9.6              pyhd3eb1b0_0
jsonschema                4.4.0              pyhd8ed1ab_0    conda-forge
jupyter                   1.0.0            py37hf985489_7    conda-forge
jupyter_client            7.2.1              pyhd8ed1ab_0    conda-forge
jupyter_console           6.4.3              pyhd8ed1ab_0    conda-forge
jupyter_core              4.9.2            py37hf985489_0    conda-forge
jupyter_server            1.16.0             pyhd8ed1ab_0    conda-forge
jupyterlab                3.3.2              pyhd8ed1ab_0    conda-forge
jupyterlab_pygments       0.1.2              pyh9f0ad1d_0    conda-forge
jupyterlab_server         2.12.0             pyhd8ed1ab_0    conda-forge
jupyterlab_widgets        1.1.0              pyhd8ed1ab_0    conda-forge
jxrlib                    1.1                  h35c211d_2    conda-forge
kealib                    1.4.14               ha22a8b1_3    conda-forge
kiwisolver                1.4.2            py37h18621fa_0    conda-forge
krb5                      1.19.3               hb49756b_0    conda-forge
lcms2                     2.12                 h577c468_0    conda-forge
lerc                      3.0                  he49afe7_0    conda-forge
libaec                    1.0.6                he49afe7_0    conda-forge
libarchive                3.5.2                h2b60450_1    conda-forge
libblas                   3.9.0           13_osx64_openblas    conda-forge
libbrotlicommon           1.0.9                h0d85af4_6    conda-forge
libbrotlidec              1.0.9                h0d85af4_6    conda-forge
libbrotlienc              1.0.9                h0d85af4_6    conda-forge
libcblas                  3.9.0           13_osx64_openblas    conda-forge
libclang                  13.0.1          default_he082bbe_0    conda-forge
libcrc32c                 1.1.2                he49afe7_0    conda-forge
libcurl                   7.82.0               h9f20792_0    conda-forge
libcxx                    13.0.1               hc203e6f_0    conda-forge
libdap4                   3.20.6               h3e144a0_2    conda-forge
libdeflate                1.10                 h0d85af4_0    conda-forge
libedit                   3.1.20210910         hca72f7f_0
libev                     4.33                 haf1e3a3_1    conda-forge
libevent                  2.1.10               h815e4d9_4    conda-forge
libffi                    3.4.2                h0d85af4_5    conda-forge
libgdal                   3.4.0               h3ab75b0_12    conda-forge
libgfortran               5.0.0           9_3_0_h6c81a4c_23    conda-forge
libgfortran5              9.3.0               h6c81a4c_23    conda-forge
libglib                   2.70.2               hf1fb8c0_4    conda-forge
libgoogle-cloud           1.35.0               h52a585b_2    conda-forge
libiconv                  1.16                 haf1e3a3_0    conda-forge
libkml                    1.3.0             h8fd9edb_1014    conda-forge
liblapack                 3.9.0           13_osx64_openblas    conda-forge
liblief                   0.11.5               he49afe7_1    conda-forge
libllvm11                 11.1.0               hd011deb_3    conda-forge
libllvm13                 13.0.1               h64f94b2_2    conda-forge
libnetcdf                 4.8.1           nompi_h6609ca0_101    conda-forge
libnghttp2                1.47.0               h942079c_0    conda-forge
libopenblas               0.3.18          openmp_h3351f45_0    conda-forge
libpng                    1.6.37               h7cec526_2    conda-forge
libpq                     14.2                 hea3049e_0    conda-forge
libprotobuf               3.19.4               hcf210ce_0    conda-forge
librttopo                 1.1.0                h0d74fd6_8    conda-forge
libsodium                 1.0.18               hbcb3906_1    conda-forge
libspatialindex           1.9.3                he49afe7_4    conda-forge
libspatialite             5.0.1               h24415ee_12    conda-forge
libssh2                   1.10.0               h52ee1ee_2    conda-forge
libthrift                 0.15.0               hab56fdc_1    conda-forge
libtiff                   4.3.0                h17f2ce3_3    conda-forge
libutf8proc               2.7.0                h0d85af4_0    conda-forge
libwebp                   1.2.2                h28dabe5_0    conda-forge
libwebp-base              1.2.2                h0d85af4_1    conda-forge
libxcb                    1.13              h0d85af4_1004    conda-forge
libxml2                   2.9.12               h7e28ab6_1    conda-forge
libzip                    1.8.0                h8b0c345_1    conda-forge
libzlib                   1.2.11            h6c3fc93_1014    conda-forge
libzopfli                 1.0.3                h046ec9c_0    conda-forge
llvm-openmp               13.0.1               hcb1a161_1    conda-forge
llvmlite                  0.38.0           py37h62ea057_0    conda-forge
locket                    0.2.1            py37hecd8cb5_2
lz4-c                     1.9.3                he49afe7_1    conda-forge
lzo                       2.10              haf1e3a3_1000    conda-forge
macfsevents               0.8.1           py37h271585c_1002    conda-forge
mapclassify               2.4.3              pyhd8ed1ab_0    conda-forge
markdown                  3.3.6              pyhd8ed1ab_0    conda-forge
markupsafe                2.1.1            py37h69ee0a8_0    conda-forge
matplotlib                3.5.1            py37hf985489_0    conda-forge
matplotlib-base           3.5.1            py37h3147e9e_0    conda-forge
matplotlib-inline         0.1.3              pyhd8ed1ab_0    conda-forge
mccabe                    0.6.1                      py_1    conda-forge
mistune                   0.8.4           py37h271585c_1005    conda-forge
msgpack-python            1.0.3            py37h737db71_0    conda-forge
multidict                 6.0.2            py37h271585c_0    conda-forge
multipledispatch          0.6.0                      py_0    conda-forge
munch                     2.5.0                      py_0    conda-forge
munkres                   1.1.4              pyh9f0ad1d_0    conda-forge
mysql-common              8.0.28               hab7e275_2    conda-forge
mysql-libs                8.0.28               h2a44bf5_2    conda-forge
nbclassic                 0.3.7              pyhd8ed1ab_0    conda-forge
nbclient                  0.5.13             pyhd8ed1ab_0    conda-forge
nbconvert                 6.4.5            py37hf985489_0    conda-forge
nbformat                  5.2.0              pyhd8ed1ab_0    conda-forge
nbsmoke                   0.6.0                      py_0    pyviz/label/dev
ncurses                   6.3                  he49afe7_0    conda-forge
nest-asyncio              1.5.4              pyhd8ed1ab_0    conda-forge
netcdf4                   1.5.8           nompi_py37h5a78667_101    conda-forge
networkx                  2.7.1              pyhd8ed1ab_0    conda-forge
notebook                  6.4.10             pyha770c72_0    conda-forge
notebook-shim             0.1.0              pyhd8ed1ab_0    conda-forge
nspr                      4.33                 he9d5cce_0
nss                       3.76                 hfce436b_0    conda-forge
numba                     0.55.1           py37h078fc1e_0    conda-forge
numcodecs                 0.9.1            py37hd8d24ac_2    conda-forge
numpy                     1.21.5           py37h3c8089f_0    conda-forge
openjpeg                  2.4.0                h6e7aa92_1    conda-forge
openssl                   1.1.1n               h6c3fc93_0    conda-forge
orc                       1.7.3                h84518c8_0    conda-forge
outcome                   1.1.0              pyhd8ed1ab_0    conda-forge
packaging                 21.3               pyhd8ed1ab_0    conda-forge
pandas                    1.3.5            py37h5b83a90_0    conda-forge
pandoc                    2.17.1.1             h694c41f_0    conda-forge
pandocfilters             1.5.0              pyhd8ed1ab_0    conda-forge
panel                     0.13.0a47                  py_0    pyviz/label/dev
param                     1.12.1a1                   py_0    pyviz/label/dev
parameterized             0.8.1              pyhd3deb0d_0    conda-forge
parquet-cpp               1.5.1                         1    conda-forge
parso                     0.8.3              pyhd8ed1ab_0    conda-forge
partd                     1.2.0              pyhd8ed1ab_0    conda-forge
pcre                      8.45                 he49afe7_0    conda-forge
pexpect                   4.8.0            py37hc8dfbb8_1    conda-forge
pickleshare               0.7.5           py37hc8dfbb8_1002    conda-forge
pillow                    9.0.1            py37h2540ef4_2    conda-forge
pip                       22.0.4             pyhd8ed1ab_0    conda-forge
pixman                    0.40.0               hbcb3906_0    conda-forge
pkginfo                   1.8.2              pyhd8ed1ab_0    conda-forge
plotly                    5.6.0              pyhd8ed1ab_0    conda-forge
pluggy                    1.0.0            py37hf985489_2    conda-forge
pooch                     1.6.0              pyhd8ed1ab_0    conda-forge
poppler                   21.11.0              h9573804_0    conda-forge
poppler-data              0.4.11               hd8ed1ab_0    conda-forge
postgresql                14.2                 he8fe76e_0    conda-forge
proj                      8.2.0                h1512c50_0    conda-forge
prometheus_client         0.13.1             pyhd8ed1ab_0    conda-forge
prompt-toolkit            3.0.27             pyha770c72_0    conda-forge
prompt_toolkit            3.0.27               hd8ed1ab_0    conda-forge
psutil                    5.9.0            py37h271585c_0    conda-forge
pthread-stubs             0.4               hc929b4f_1001    conda-forge
ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
py                        1.11.0             pyh6c4a22f_0    conda-forge
py-lief                   0.11.5           py37hd8d24ac_1    conda-forge
pyarrow                   7.0.0           py37hd1ae41a_2_cpu    conda-forge
pycodestyle               2.8.0              pyhd8ed1ab_0    conda-forge
pycosat                   0.6.3           py37h271585c_1009    conda-forge
pycparser                 2.21               pyhd8ed1ab_0    conda-forge
pyct                      0.4.8                      py_0    pyviz/label/dev
pyct-core                 0.4.8                      py_0    pyviz/label/dev
pyctdev                   0.6.6                      py_0    pyviz/label/dev
pyepsg                    0.4.0                      py_0    conda-forge
pyflakes                  2.4.0              pyhd8ed1ab_0    conda-forge
pygments                  2.11.2             pyhd8ed1ab_0    conda-forge
pyopenssl                 22.0.0             pyhd8ed1ab_0    conda-forge
pyparsing                 3.0.7              pyhd8ed1ab_0    conda-forge
pyproj                    3.2.1            py37hf78b1f3_5    conda-forge
pyqt                      5.12.3           py37hf985489_8    conda-forge
pyqt-impl                 5.12.3           py37hab5ec1f_8    conda-forge
pyqt5-sip                 4.19.18          py37h070e122_8    conda-forge
pyqtchart                 5.12             py37hab5ec1f_8    conda-forge
pyqtwebengine             5.12.1           py37hab5ec1f_8    conda-forge
pyrsistent                0.18.1           py37h271585c_0    conda-forge
pyshp                     2.2.0              pyhd8ed1ab_0    conda-forge
pysocks                   1.7.1            py37hf985489_4    conda-forge
pytest                    7.1.1            py37hf985489_0    conda-forge
pytest-cov                3.0.0              pyhd8ed1ab_0    conda-forge
python                    3.7.12          haf480d7_100_cpython    conda-forge
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
python-libarchive-c       4.0              py37hf985489_0    conda-forge
python-snappy             0.6.0            py37haf2aeb7_1    conda-forge
python_abi                3.7                     2_cp37m    conda-forge
pytz                      2022.1             pyhd8ed1ab_0    conda-forge
pyviz_comms               2.2.0                      py_0    pyviz/label/dev
pywavelets                1.3.0            py37h4105427_0    conda-forge
pyyaml                    6.0              py37h271585c_3    conda-forge
pyzmq                     22.3.0           py37h8f778e5_1    conda-forge
qt                        5.12.9               h2a607e2_5    conda-forge
qtconsole                 5.3.0              pyhd8ed1ab_0    conda-forge
qtconsole-base            5.3.0              pyhd8ed1ab_0    conda-forge
qtpy                      2.0.1              pyhd8ed1ab_0    conda-forge
rasterio                  1.2.10           py37h8e2be48_3    conda-forge
re2                       2022.02.01           he49afe7_0    conda-forge
readline                  8.1.2                hca72f7f_1
requests                  2.27.1             pyhd8ed1ab_0    conda-forge
retrying                  1.3.3                      py_2    conda-forge
ripgrep                   13.0.0               hbbacdb1_2    conda-forge
rtree                     0.9.7            py37hf13911c_3    conda-forge
ruamel_yaml               0.15.100         py37h9ed2024_0
s3fs                      2022.2.0           pyhd8ed1ab_1    conda-forge
scikit-image              0.19.2           py37h5b83a90_0    conda-forge
scikit-learn              1.0.2            py37h572704e_0    conda-forge
scipy                     1.7.3            py37h4e3cf02_0    conda-forge
selenium                  4.1.0              pyhd8ed1ab_0    conda-forge
send2trash                1.8.0              pyhd8ed1ab_0    conda-forge
setuptools                59.8.0           py37hf985489_0    conda-forge
shapely                   1.8.0            py37hd43bd85_4    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
snappy                    1.1.8                hb1e8313_3    conda-forge
sniffio                   1.2.0            py37hf985489_2    conda-forge
snuggs                    1.4.7                      py_0    conda-forge
sortedcontainers          2.4.0              pyhd8ed1ab_0    conda-forge
soupsieve                 2.3.1              pyhd8ed1ab_0    conda-forge
spatialpandas             0.4.3                      py_0    pyviz/label/dev
sqlite                    3.37.2               h707629a_0
streamz                   0.6.3              pyh6c4a22f_0    conda-forge
tblib                     1.7.0              pyhd8ed1ab_0    conda-forge
tenacity                  8.0.1              pyhd8ed1ab_0    conda-forge
terminado                 0.13.3           py37hf985489_0    conda-forge
testpath                  0.6.0              pyhd8ed1ab_0    conda-forge
threadpoolctl             3.1.0              pyh8a188c0_0    conda-forge
tifffile                  2022.3.25          pyhd8ed1ab_0    conda-forge
tiledb                    2.5.3                hca4410a_0    conda-forge
tk                        8.6.12               h5dbffcc_0    conda-forge
toml                      0.10.2             pyhd8ed1ab_0    conda-forge
tomli                     2.0.1              pyhd8ed1ab_0    conda-forge
toolz                     0.11.2             pyhd8ed1ab_0    conda-forge
tornado                   6.1              py37h271585c_2    conda-forge
tqdm                      4.63.1             pyhd8ed1ab_0    conda-forge
traitlets                 5.1.1              pyhd8ed1ab_0    conda-forge
trio                      0.20.0           py37hf985489_0    conda-forge
trio-websocket            0.9.2              pyhd8ed1ab_0    conda-forge
typing-extensions         4.1.1                hd8ed1ab_0    conda-forge
typing_extensions         4.1.1              pyha770c72_0    conda-forge
tzcode                    2022a                h5eb16cf_0    conda-forge
tzdata                    2022a                h191b570_0    conda-forge
unicodedata2              14.0.0           py37h271585c_0    conda-forge
urllib3                   1.26.9             pyhd8ed1ab_0    conda-forge
wcwidth                   0.2.5              pyh9f0ad1d_2    conda-forge
webencodings              0.5.1                      py_1    conda-forge
websocket-client          1.3.2              pyhd8ed1ab_0    conda-forge
wheel                     0.37.1             pyhd8ed1ab_0    conda-forge
widgetsnbextension        3.6.0            py37hf985489_0    conda-forge
wrapt                     1.14.0           py37h69ee0a8_0    conda-forge
wsproto                   1.1.0            py37hf985489_0    conda-forge
xarray                    0.20.2             pyhd8ed1ab_0    conda-forge
xerces-c                  3.2.3                h6564042_4    conda-forge
xorg-libxau               1.0.9                h35c211d_0    conda-forge
xorg-libxdmcp             1.1.3                h35c211d_0    conda-forge
xyzservices               2022.3.0           pyhd8ed1ab_0    conda-forge
xz                        5.2.5                haf1e3a3_1    conda-forge
yaml                      0.2.5                h0d85af4_2    conda-forge
yarl                      1.7.2            py37h271585c_1    conda-forge
zarr                      2.11.1             pyhd8ed1ab_0    conda-forge
zeromq                    4.3.4                he49afe7_1    conda-forge
zfp                       0.5.5                h4a89273_8    conda-forge
zict                      2.1.0              pyhd8ed1ab_0    conda-forge
zipp                      3.7.0              pyhd8ed1ab_1    conda-forge
zlib                      1.2.11            h6c3fc93_1014    conda-forge
zstd                      1.5.2                h582d3a0_0    conda-forge
```
</details>

philippjfr replied:
Thanks will try too.

philippjfr replied:
Yeah, can't reproduce at all on my M1, tried python 3.8, 3.9 and python 3.10 environments and pandas 1.3.5 and 1.4.1

philippjfr replied:
Fixed by upgrading pandas and by this PR https://github.com/holoviz/param/pull/612

maximlt commented on examples/user_guide/Interactive.ipynb:
```diff
@@ -92,6 +123,68 @@
+   "source": [
+    "## Functions as inputs\n",
+    "\n",
+    "In some cases your starting point for your interactive pipeline may not simply be a DataFrame or xarray Dataset but a function that fetches some data or applies some initial processing on your data. In such a case you can use the `hvplot.bind` function to bind static AND dynamic arguments to your function. Binding dynamic arguments such as a widget or parameter means that whenever the widget ...
```
It'd be nice to include in the example a static argument. And I assume that when you mention 'parameters' you refer to param's Parameters? If so I guess there should be a short intro about them and a link to param, as long as with an example. I can do these two things if you agree they should go in.

maximlt commented on hvplot/__init__.py:
```diff
@@ -127,6 +130,13 @@ def _patch_doc(cls, kind):
         method.__signature__ = signature
 
 
+@_wraps(_pn.bind)
```
I'm not sure Panel's docstring for `bind` will be very understandable by hvPlot users.

maximlt commented on examples/user_guide/Interactive.ipynb:
```diff
@@ -92,6 +123,68 @@
+   "source": [
+    "## Functions as inputs\n",
+    "\n",
+    "In some cases your starting point for your interactive pipeline may not simply be a DataFrame or xarray Dataset but a function that fetches some data or applies some initial processing on your data. In such a case you can use the `hvplot.bind` function to bind static AND dynamic arguments to your function. Binding dynamic arguments such as a widget or parameter means that whenever the widget ...
```
What return types are supported? Dask and Pandas Dataframes and Xarray Datasets?

philippjfr replied:
gpd.DataFrame, cudf.DataFrame, dask.dataframe.DataFrame, pandas.DataFrame.

maximlt commented on hvplot/interactive.py:
```diff
@@ -52,11 +55,35 @@ class Interactive():
+    def applies(cls, obj):
+        return True
+
+    def __init__(self, obj, transform=None, fn=None, plot=False, depth=0,
```
Not related to this PR, but it'd be nice to have a docstring here.

maximlt reviewed (commented):
I'd say there is one outstanding issue to fix, my other comments or suggestions are somewhat minor.

philippjfr commented on examples/user_guide/Interactive.ipynb:
```diff
@@ -92,6 +123,68 @@
+    "\n",
+    "dt_range = pn.widgets.DateRangeSlider(start=df.date.iloc[-1000], end=df.date.max(), value=(df.date.iloc[-100], df.date.max()))\n",
+    "\n",
+    "stock_dfi[(stock_dfi.date>=dt_range.param.value_start) & (stock_dfi.date<=dt_range.param.value_end)].hvplot(kind='ohlc', grid=True, title=ticker)"
```
This was caused by https://github.com/holoviz/panel/pull/3276

philippjfr commented:
Sorry for polluting this PR but I pushed some other changes:

- Accessor resolution now implemented consistently via one codepath
- Accessors now always return a new interactive object, previously it would act by changing the object inplace resulting in a lot of oddness when you chained that object
- Fixed docstring generation for methods
